### ISSUE TITLE
Add code for generating generic concept-based cohorts?

### ISSUE THREAD

**Issue by schuemie** [OHDSI/CohortGenerator#9]
For our negative controls we tend to use simple cohorts based on concepts alone. For example, in our ATLAS estimation module, you can choose to generate cohorts per concept, for a list of user-defined concepts, with these options:

1. All occurrences or first?
2. Include descendants?

I wonder if it would make sense to include the code for generating these cohorts in `CohortGenerator`, just to have all cohort generation code in one place. Currently this code lives in the estimation skeleton, with for example the SQL [here](https://github.com/OHDSI/SkeletonComparativeEffectStudy/blob/main/inst/sql/sql_server/NegativeControlOutcomes.sql).

**Comment by anthonysena** [comment]
I'd support this and to add to this particular issue: there is also code for building negative control cohorts in Circe: https://github.com/OHDSI/circe-be/tree/master/src/main/java/org/ohdsi/circe/cohortdefinition/negativecontrols. Tagging @chrisknoll since perhaps we should deprecate that code in favor of having this type of cohort generation directly in CohortGenerator instead?

**Comment by anthonysena** [comment]
To follow up on this topic and to make some notes on how we could go about this implementation:

- The `OutcomeCohortExpression` used in circe-be [here](https://github.com/OHDSI/circe-be/blob/master/src/main/java/org/ohdsi/circe/cohortdefinition/negativecontrols/OutcomeCohortExpression.java) would allow for storing the `occurrenceType` (All/First), `detectOnDescendants` (TRUE/FALSE), and the `domains` (currently the skeleton code referenced above uses condition_occurrence but we can provide some flexibility here). We'd want to replicate this data structure in CohortGenerator.
- Currently, the `OutcomeCohortExpression` lacks the list of concepts - these are presumed to be in the form of a concept set expression stored separately. One potential challenge here is the resolution of the concept set into a list of concepts so perhaps we should just opt for that. Someone calling CohortGenerator should be able to go from a concept set to a list of concepts...but maybe that's not easily done via CirceR? Tagging @chrisknoll here.
- The code for generating the negative control SQL is [here](https://github.com/OHDSI/circe-be/blob/master/src/main/java/org/ohdsi/circe/cohortdefinition/negativecontrols/CohortExpressionQueryBuilder.java)

So I think a path forward here would be: create a data structure in CohortGenerator for defining negative control outcomes that includes the `occurrenceType`, `detectOnDescendants` and `domains` and a list of `conceptIds` for the outcomes. CohortGenerator would take this data structure, build and execute the SQL to produce the negative control outcome cohorts.

The preliminary data structure would look like:

- `occurrenceType` - string with choice of "ALL" or "FIRST"
- `detectOnDescendants` - boolean TRUE/FALSE 
- `domains` - vector of strings defined by:  condition, drug, device, measurement, observation, procedure, visit (basing this on circe-be)
- `conceptIds` - vector of bigints that are used to define the negative control outcomes. Each conceptId will be used as the cohort_definition_id for the negative control outcome.

Tagging @schuemie for awareness and any feedback.

**Comment by gowthamrao** [comment]
This would be very valuable. 

In a recent study, i had to set generateCohorts = FALSE in the estimation package generated by Atlas Skeleton. This was because i wanted to reuse the T/C/O cohorts that were already instantiated. But while doing so - I did not get negative control distribution, because negative control cohorts were not generated. 

I had to then copy snipped of the negative control generation process from the skeleton and run them outside. It was ok- but if cohortGenerator could handle it - it would be elegant.
